spec/controllers/staffs_controller_spec.rb

```diff
@@ -0,0 +1,228 @@
+
+  describe "not signin" do
+    context "GET #index" do
+      before {get :index}
+
+      it "should found" do
+        expect(response).to have_http_status(302)
+      end
+
+      it "redirect to signin page" do
+        expect(response).to redirect_to signin_path
+      end
+    end
+
+    context "GET #new" do
+      before {get :new}
+
+      it "redirect to signin page" do
+        expect(response).to redirect_to signin_path
+      end
+
+      it "should found" do
+        expect(response).to have_http_status(302)
+      end
+    end
+
+    context "POST #create" do
+      before do
+        post :create, params: {staff: {user_name: "yen"}}
+      end
+
+      it "redirect to signin page" do
+        expect(response).to redirect_to signin_path
+      end
+
+      it "should found" do
+        expect(response).to have_http_status(302)
+      end
+    end
+
+    context "GET #edit" do
+      before do
+        get :edit, params: {id: 1}
+      end
+
+      it "redirect to signin page" do
+        expect(response).to redirect_to signin_path
+      end
+
+      it "should found" do
+        expect(response).to have_http_status(302)
+      end
+    end
+
+    context "PATCH #update" do
+      before do
+        patch :update, params: {id: 1}
+      end
+
+      it "redirect to signin page" do
+        expect(response).to redirect_to signin_path
+      end
+
+      it "should found" do
+        expect(response).to have_http_status(302)
+      end
+    end
+
+    context "DELETE #destroy" do
+      before do
+        delete :destroy, params: {id: 1}
+      end
+
+      it "redirect to signin page" do
+        expect(response).to redirect_to signin_path
+      end
+
+      it "should found" do
+        expect(response).to have_http_status(302)
+      end
+    end
+  end
+
+  describe "signin with admin account" do
+    before do
+      @admin = FactoryBot.create :user, role: "Admin"
+      log_in @admin
+    end
+
+    context "GET #index" do
+      before do
+        get :index
+      end
+
+      it "should be success" do
+        expect(response).to have_http_status(200)
+      end
+
+      it "render index template" do
+        expect(response).to render_template :index
+      end
+
+      it "assigns @staffs to staffs list" do
+        expect(assigns(:staffs)).to match_array staffs
+      end
+    end
+
+    context "GET #new" do
+      before do
+        get :new
+      end
+
+      it "should be success" do
+        expect(response).to have_http_status(200)
+      end
+
+      it "render new template" do
+        expect(response).to render_template :new
+      end
+
+      it "assigns @staff to a new staff" do
+        expect(assigns(:staff)).to be_a_new Staff
+      end
+    end
+
+    context "POST #create" do
+      context "with valid attributes" do
+        before do
+          post :create, params: {staff: FactoryBot.attributes_for(:staff).merge(user_name: "yen", password: "111111", password_confirmation: "111111")}
```

Comment:
Không dùng toàn bộ là Factory được hả em?

Reply:
phía dưới em kiểm tra giá trị của user_name gán vào nên em ghi đè nó ạ